design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,163 @@
+
+### Proposed Implementations
+
+- CStor-pool Manger inbuilt CSPI controller watches for CSPI resources and tries to get the desired state via every sync operation. It will iterate over all the BlockDevices that exist on CStorPoolInstance resource and if BlockDevice resource capacity(considerable changes) is greater than the existing capacity of BlockDevice on CstorPoolInstanceBlockDevice then we can follow below-mentioned steps to perform the pool expansion:
+1. If IsBlockDeviceExpanded flag sets true then it is something like user/admin is requesting to perform pool expansion steps.
+2. If IsBlockDeviceExpanded flag sets false then NoOps will be performed.
+[What if cStor is consuming part of partitioned disk? User has to set IsBlockDeviceExpanded to true and other part of partitioned disk shouldn't have any FileSystem then only pool expansion will be triggered].
+[What if cStor is consuming entier partitioned disk? User has to set IsBlockDeviceExpanded to true then cstor-pool-mgmt will expand the partition and trigger pool expansion process].
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+```
+
+To get user Input proposed to add field in above struct
+
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+
+	// IsBlockDeviceExpanded informs whether blockdevice 
+	// got expanded or not
+	IsBlockDeviceExpanded bool `json:”isBlockDeviceExpanded”`
+}
+```
+
+### Steps to perform user stories
+- User has to set the value of IsBlockDeviceExpanded to true in corresponding CStorPoolInstanceBlockDevice.
+- Once the operation is completed user can unset the value of ISBlockDeviceExpanded[How users can identify
+  operation is succeeded? User can know it via events (or) Increased capacity in CStorPoolInstanceBlockDevice].
+  NOTE: Leaving IsBlockDeviceExpanded can be left set it will not cause any extra network calls.
+
+#### Pool Expansion Steps:
+
+Step1: Set autoexpand to on in pool by triggering `zpool set autoexpand=on <pool_name>`.
```

Comment:
can this be set already during pool creation?

Reply:
Yes!!